File: src/test/java/no/nav/fo/veilarbregistrering/enhet/KommunenummerTest.java
package no.nav.fo.veilarbregistrering.enhet;

import org.junit.jupiter.api.Test;

import static no.nav.fo.veilarbregistrering.enhet.Kommunenummer.KommuneMedBydel.*;
import static org.assertj.core.api.Assertions.assertThat;

public class KommunenummerTest {

    @Test
    public void oslo_er_en_kommune_med_flere_bydeler() {
        Kommunenummer osloKommune = Kommunenummer.of(OSLO);
        assertThat(osloKommune.kommuneMedBydeler()).isTrue();
    }

    @Test
    public void bergen_er_en_kommune_med_flere_bydeler() {
        Kommunenummer bergenKommune = Kommunenummer.of(BERGEN);
        assertThat(bergenKommune.kommuneMedBydeler()).isTrue();
    }

    @Test
    public void trondheim_er_en_kommune_med_flere_bydeler() {
        Kommunenummer trondheimKommune = Kommunenummer.of(TRONDHEIM);
        assertThat(trondheimKommune.kommuneMedBydeler()).isTrue();
    }

    @Test
    public void stavanger_er_en_kommune_med_flere_bydeler() {
        Kommunenummer stavangerKommune = Kommunenummer.of(STAVANGER);
        assertThat(stavangerKommune.kommuneMedBydeler()).isTrue();
    }

    @Test
    public void horten_er_ikke_en_kommune_med_flere_bydeler() {
        Kommunenummer hortenKommune = Kommunenummer.of("3801");
        assertThat(hortenKommune.kommuneMedBydeler()).isFalse();
    }
}

File: src/test/java/no/nav/fo/veilarbregistrering/enhet/KommunenummerTest.kt
package no.nav.fo.veilarbregistrering.enhet

import no.nav.fo.veilarbregistrering.enhet.Kommunenummer.KommuneMedBydel
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class KommunenummerTest {

    @Test
    fun `oslo er en kommune med flere bydeler`() {
        val osloKommune = Kommunenummer.of(KommuneMedBydel.OSLO)
        assertThat(osloKommune.kommuneMedBydeler()).isTrue
    }

    @Test
    fun `bergen er en kommune med flere bydeler`() {
        val bergenKommune = Kommunenummer.of(KommuneMedBydel.BERGEN)
        assertThat(bergenKommune.kommuneMedBydeler()).isTrue
    }

    @Test
    fun `trondheim er en kommune med flere bydeler`() {
        val trondheimKommune = Kommunenummer.of(KommuneMedBydel.TRONDHEIM)
        assertThat(trondheimKommune.kommuneMedBydeler()).isTrue
    }

    @Test
    fun `stavanger er en kommune med flere bydeler`() {
        val stavangerKommune = Kommunenummer.of(KommuneMedBydel.STAVANGER)
        assertThat(stavangerKommune.kommuneMedBydeler()).isTrue
    }

    @Test
    fun `horten er ikke en kommune med flere bydeler`() {
        val hortenKommune = Kommunenummer.of("3801")
        assertThat(hortenKommune.kommuneMedBydeler()).isFalse
    }
}
